### PR TITLE
[WIP] Plan Convertion from Presto to Omnisci: part-I

### PR DESCRIPTION
**Pull request by yma11**

### CODE REVIEW

**Comment by yma11** [comment]
@winningsix @jikunshang Please take a look on the structure of PlanNodes and Expressions. Currently toJson is not implemented yet.

File: java/plan/src/main/java/com/mapd/plan/CiderCallExpression.java [hunk 1]
```diff
@@ -37,13 +37,15 @@ public CiderFilterCondition(String op, String type, List<CiderExpression> expres
...
 
+    @Override
     public ObjectNode toJson(ObjectMapper objectMapper)
     {
         ObjectNode conditionNode = objectMapper.createObjectNode();
         conditionNode.put("op", matchOperator(op));
         ArrayNode operandsNode = objectMapper.createArrayNode();
         ObjectNode operandsFeilds = objectMapper.createObjectNode();
         ObjectNode operandsInput = objectMapper.createObjectNode();
+        /*
```
Comment (winningsix) [inline java/plan/src/main/java/com/mapd/plan/CiderCallExpression.java:48]: Remove this if not needed.

File: java/plan/src/main/java/com/mapd/plan/CiderConstantExpression.java [hunk 2]
```diff
@@ -39,6 +40,11 @@ private int getIntLiteral(int scale, double literal)
     }
 
     @Override
+    public ObjectNode toJson(ObjectMapper objectManager)
+    {
+        return null;
```
Comment (winningsix) [inline java/plan/src/main/java/com/mapd/plan/CiderConstantExpression.java:45]: Throw exception please.

File: java/plan/src/main/java/com/mapd/plan/CiderSpecialFormCondition.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        this.type = type;
+        this.conditions = conditions;
+    }
+
+    @Override
+    public ObjectNode toJson(ObjectMapper objectMapper)
+    {
+        return null;
+    }
+    public enum Form
```
Comment (winningsix) [inline java/plan/src/main/java/com/mapd/plan/CiderSpecialFormCondition.java:32]: You mean IDENTIFIER or TOKEN_IDENTIFIER?

File: java/plan/src/main/java/com/mapd/plan/CiderVariableReferenceExpression.java [hunk 4]
```diff
@@ -13,27 +13,80 @@
...
     {
         // column index
         operandsInput.put("input", columnIndex);
-        String dataType = matchType(targetType);
         operandsFeilds.put("target_type", matchType(targetType).toUpperCase());
         operandsFeilds.put("type_scale", getTypeScale(targetType));
         operandsFeilds.put("type_precision", getTypePrecision(targetType));
     }
+
+    protected String matchType(String type)
```
Comment (winningsix) [inline java/plan/src/main/java/com/mapd/plan/CiderVariableReferenceExpression.java:54]: why do we need to move from super class here?

**Review by winningsix — COMMENTED** [review]
Hmm. current part is Prestodb free impl, right? All classes are based on OmnisciDB concept. It's more like a SDK to upper layer engine. If so, please rename the PR title avoiding misleading.

Overall LGTM. I didn't have sanity check on the correctness.

**Comment by winningsix** [comment]
@yma11  Intel-bigdata:p-cleanup  Please try to base on this branch after @xieqi updates the dependency

File: java/plan/src/main/java/com/mapd/plan/CiderVariableReferenceExpression.java (hunk 4 above)
Comment (yma11) [reply java/plan/src/main/java/com/mapd/plan/CiderVariableReferenceExpression.java:54]: actually this method should not be used by every subclass. Will put it back if needed.

File: java/plan/src/main/java/com/mapd/plan/CiderSpecialFormCondition.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
+package com.mapd.plan;
```
Comment (jikunshang) [inline java/plan/src/main/java/com/mapd/plan/CiderSpecialFormCondition.java:1]: license

File: java/plan/src/main/java/com/mapd/plan/CiderProjectNode.java [hunk 6]
```diff
@@ -18,21 +18,32 @@
...
+import java.util.HashMap;
 import java.util.List;
 
 public class CiderProjectNode
         extends CiderOperatorNode
 {
+    // TODO: remove fieldNames and exprs
     private List<String> fieldNames;
     private List<Pair<String, Integer>> exprs;
+    private HashMap<CiderVariableReferenceExpression, CiderExpression> assignments;
```
Comment (jikunshang) [inline java/plan/src/main/java/com/mapd/plan/CiderProjectNode.java:30]: why need a hashmap here?

File: java/plan/src/main/java/com/mapd/plan/CiderSpecialFormCondition.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static java.util.Collections.unmodifiableList;
+
+public class CiderSpecialFormCondition extends CiderExpression {
```
Comment (jikunshang) [inline java/plan/src/main/java/com/mapd/plan/CiderSpecialFormCondition.java:11]: why need this class? can you add some comments?

**Review by jikunshang — COMMENTED** [review]
can we maintain a design doc or class diagram for better understanding?

**Comment by winningsix** [comment]
Closed per communicated.

**Pull request closed without merging**